is.js

```diff
@@ -399,7 +399,17 @@
 
     // is a given string palindrome?
     is.palindrome = function(str) {
-        return is.string(str) && str === str.split('').reverse().join('');
+        if (is.not.string(str)) {
+            return false;
+        }
+        var str = str.replace(/[^a-zA-Z0-9]+/gi, '').toLowerCase();
+        var len = str.length - 1;
```

Comment:
May I know the reason?
I mean is it for maintaining the proper code or something else.?